concordium-consensus/src/Concordium/Scheduler.hs

```diff
@@ -335,36 +330,38 @@ dispatch (msg, mVerRes) = do
                      handleTransferWithSchedule (mkWTC TTTransferWithScheduleAndMemo) twswmTo twswmSchedule $ Just twswmMemo
 
                    ConfigureBaker{..} ->
-                     onlyAccountVersion SAccountV1 $
-                     onlyChainParametersVersion SCPV1 $
+                     onlyWithDelegation $
                      handleConfigureBaker (mkWTC TTConfigureBaker) cbCapital cbRestakeEarnings cbOpenForDelegation cbKeysWithProofs cbMetadataURL cbTransactionFeeCommission cbBakingRewardCommission cbFinalizationRewardCommission
 
                    ConfigureDelegation{..} ->
-                     onlyAccountVersion SAccountV1 $
-                     onlyChainParametersVersion SCPV1 $
+                     onlyWithDelegation $
                      handleConfigureDelegation (mkWTC TTConfigureDelegation) cdCapital cdRestakeEarnings cdDelegationTarget
 
           case res of
             -- The remaining block energy is not sufficient for the handler to execute the transaction.
             Nothing -> return Nothing
             Just summary -> return $ Just $ TxValid summary
   where
-    -- Function 'onlyAccountVersion' @sav@ @k@ fails if account version for @MPV m@ is not that of @sav@,
-    -- and continues with @k@ if they are the same.
-    onlyAccountVersion :: SAccountVersion av -> ((AccountVersionFor (MPV m) ~ av) => a) -> a
-    onlyAccountVersion sav c =
-        case (sav, accountVersionFor (protocolVersion @(MPV m))) of
-            (SAccountV0, SAccountV0) -> c
-            (SAccountV1, SAccountV1) -> c
-            _ -> error "Operation unsupported for this protocol version."
-    -- Function 'onlyChainParametersVersion' @scpv@ @k@ fails if chain parameters version for @MPV m@ is
-    -- not that of @sav@, and continues with @k@ if they are the same.
-    onlyChainParametersVersion :: SChainParametersVersion cpv -> ((ChainParametersVersionFor (MPV m) ~ cpv) => a) -> a
-    onlyChainParametersVersion scpv c =
-        case (scpv, chainParametersVersionFor (protocolVersion @(MPV m))) of
-            (SCPV0, SCPV0) -> c
-            (SCPV1, SCPV1) -> c
-            _ -> error "Operation unsupported for this chain parameter version."
+    -- Function @onlyWithoutDelegation k@ fails if the protocol version @MPV m@ supports
+    -- delegation. Otherwise, it continues with @k@, which may assume the chain parameters version
+    -- is 'ChainParametersV0' and the account version is 'AccountV0'.
+    onlyWithoutDelegation :: (
+        (
+          ChainParametersVersionFor (MPV m) ~ 'ChainParametersV0,
+          AccountVersionFor (MPV m) ~ 'AccountV0
+        ) => a)
+        -> a
+    onlyWithoutDelegation c = case protocolVersion @(MPV m) of
+      SP1 -> c
+      SP2 -> c
+      SP3 -> c
+      _ -> error "Operation unsupported at this prtocol version."
```

Comment:
Could this be done by casing on `delegationSupport @(AccountVersionFor (MPV m))` like the function just below?

Reply:
Unfortunately not. The constraint on the chain parameters version cannot be inferred just from knowing that the account version is `AccountV0`.